lib/hotel.rb

```diff
@@ -0,0 +1,189 @@
+
+    attr_reader :rooms, :price, :reservation_collection, :block_reservation_collection, :reservation_made
+
+    def initialize(num_of_rooms, price)
+      @rooms = []
+      room_num = 0
+      num_of_rooms.times do |room|
+        room_num += 1
+        @rooms << room_num
+      end #loop
+
+      @price = price
+      @reservation_collection = []
+      @block_reservation_collection = []
+    end #initialize
+
+    def make_reservation(check_in, check_out, room_num)
+      if room_availability(check_in, check_out).include?(room_num)
+        @reservation_made = Reservation.new(check_in, check_out, room_num)
+        @reservation_collection << @reservation_made
+      else
+        raise ArgumentError.new("It appears the room you requested is booked during that date range")
+      end
+      # @reservation_collection << @reservation_made
+    end
+
+
+    def make_block_reservation(check_in, check_out, num_of_rooms)
+      available_rooms = room_availability(check_in, check_out)
+      if available_rooms.length >= num_of_rooms
+        block_reservation = BlockRoom.new(check_in, check_out, num_of_rooms)
+        block_reservation.block_of_rooms << available_rooms.pop(num_of_rooms)
+        @block_reservation_collection << block_reservation
+        return block_reservation.block_of_rooms
+      else
+        return raise ArgumentError.new("We do not have enough rooms to reserve a block")
+      end
+    end
+
+    # ability to reserve a room within a block of rooms
+    #needs to match the date range of the block
+    def reserve_room_in_block(check_in, check_out, num_of_rooms)
+
+      check_in = Date.parse(check_in)
+      check_out = Date.parse(check_out)
+
+      @block_reservation_collection.each do |entry|
+        if check_in == entry.check_in && check_out == entry.check_out
+          if entry.available_rooms[0].length >= num_of_rooms
+            num_of_rooms.times do
+              entry.booked_rooms << entry.available_rooms[0].pop
+            end
+            return entry.booked_rooms
+          end
+        end
+      end
+    end
+
+    def blocked_rooms_availability(check_in, check_out)
+      check_in = Date.parse(check_in)
+      check_out = Date.parse(check_out)
+
+      @block_reservation_collection.each do |entry|
+        if check_in == entry.check_in && check_out == entry.check_out
+          if entry.available_rooms[0].length >= 0
+            return true
+          else false
+          end
+        end
+      end
+    end
+
+    def date_list_of_reservations(date)
+      date_list = []
+      date = Date.parse(date)
+      @reservation_collection.each do |entry|
+        # binding.pry
+        if entry.check_in <= date && entry.check_out >= date
+          date_list << [entry.room_num, entry.check_in.to_s, entry.check_out.to_s]
+        end
+      end
+      return date_list
+    end #def
+
+    def room_availability(check_in, check_out)
+      #can be string with dates
+      rooms_available = @rooms.clone
+
```

Comment:
I like that you've made getting a list of available rooms its own method - this both DRYs up your code and makes testing easier.